{
    "url": "https://api.github.com/repos/spack/spack/issues/719",
    "repository_url": "https://api.github.com/repos/spack/spack",
    "labels_url": "https://api.github.com/repos/spack/spack/issues/719/labels{/name}",
    "comments_url": "https://api.github.com/repos/spack/spack/issues/719/comments",
    "events_url": "https://api.github.com/repos/spack/spack/issues/719/events",
    "html_url": "https://github.com/spack/spack/issues/719",
    "id": 145288882,
    "node_id": "MDU6SXNzdWUxNDUyODg4ODI=",
    "number": 719,
    "title": "[BUG] No RPATH in Python extensions",
    "user": {
        "login": "citibeth",
        "id": 1924215,
        "node_id": "MDQ6VXNlcjE5MjQyMTU=",
        "avatar_url": "https://avatars.githubusercontent.com/u/1924215?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/citibeth",
        "html_url": "https://github.com/citibeth",
        "followers_url": "https://api.github.com/users/citibeth/followers",
        "following_url": "https://api.github.com/users/citibeth/following{/other_user}",
        "gists_url": "https://api.github.com/users/citibeth/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/citibeth/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/citibeth/subscriptions",
        "organizations_url": "https://api.github.com/users/citibeth/orgs",
        "repos_url": "https://api.github.com/users/citibeth/repos",
        "events_url": "https://api.github.com/users/citibeth/events{/privacy}",
        "received_events_url": "https://api.github.com/users/citibeth/received_events",
        "type": "User",
        "site_admin": false
    },
    "labels": [],
    "state": "closed",
    "locked": false,
    "assignee": null,
    "assignees": [],
    "milestone": null,
    "comments": 15,
    "created_at": "2016-04-01T21:29:33Z",
    "updated_at": "2016-07-18T04:20:39Z",
    "closed_at": "2016-07-18T04:20:39Z",
    "author_association": "MEMBER",
    "active_lock_reason": null,
    "body": "See the dialog below for details.  Since Spack does not use hierarchical modules, working around this bug requires that one do `spack load` on an arbitrary and growing set of modules.\n\n```\n[me@ankeli spack2]$ spack purge\n[me@ankeli spack2]$ spack load py-numpy\n[me@ankeli spack2]$ python3 -c 'import numpy'\nTraceback (most recent call last):\n  File \"<string>\", line 1, in <module>\n  File \"/home/me/spack2/opt/spack/linux-x86_64/gcc-4.9.3/py-numpy-1.11.0-mulodttw5pcyjufva4htsktwty4qd52r/lib/python3.5/site-packages/numpy-1.11.0-py3.5-linux-x86_64.egg/numpy/__init__.py\", line 180, in <module>\n    from . import add_newdocs\n  File \"/home/me/spack2/opt/spack/linux-x86_64/gcc-4.9.3/py-numpy-1.11.0-mulodttw5pcyjufva4htsktwty4qd52r/lib/python3.5/site-packages/numpy-1.11.0-py3.5-linux-x86_64.egg/numpy/add_newdocs.py\", line 13, in <module>\n    from numpy.lib import add_newdoc\n  File \"/home/me/spack2/opt/spack/linux-x86_64/gcc-4.9.3/py-numpy-1.11.0-mulodttw5pcyjufva4htsktwty4qd52r/lib/python3.5/site-packages/numpy-1.11.0-py3.5-linux-x86_64.egg/numpy/lib/__init__.py\", line 8, in <module>\n    from .type_check import *\n  File \"/home/me/spack2/opt/spack/linux-x86_64/gcc-4.9.3/py-numpy-1.11.0-mulodttw5pcyjufva4htsktwty4qd52r/lib/python3.5/site-packages/numpy-1.11.0-py3.5-linux-x86_64.egg/numpy/lib/type_check.py\", line 11, in <module>\n    import numpy.core.numeric as _nx\n  File \"/home/me/spack2/opt/spack/linux-x86_64/gcc-4.9.3/py-numpy-1.11.0-mulodttw5pcyjufva4htsktwty4qd52r/lib/python3.5/site-packages/numpy-1.11.0-py3.5-linux-x86_64.egg/numpy/core/__init__.py\", line 14, in <module>\n    from . import multiarray\nImportError: libopenblas.so.0: cannot open shared object file: No such file or directory\n```\n\nUh-oh, what's wrong... let's debug this with `ldd`:\n\n```\n[me@ankeli spack2]$ ldd `spack location -i py-numpy`/lib/python3.5/site-packages/numpy-1.11.0-py3.5-linux-x86_64.egg/numpy/core/multiarray.cpython-35m-x86_64-linux-gnu.so\n    linux-vdso.so.1 =>  (0x00007ffc4ef1f000)\n    libopenblas.so.0 => not found\n    libm.so.6 => /lib64/libm.so.6 (0x00007f6ccbd1d000)\n    libpython3.5m.so.1.0 => /home/me/spack2/opt/spack/linux-x86_64/gcc-4.9.3/python-3.5.1-5q5rsrtjld4u6jiicuvtnx52m7tfhegi/lib/libpython3.5m.so.1.0 (0x00007f6ccb7f7000)\n    libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f6ccb5db000)\n    libc.so.6 => /lib64/libc.so.6 (0x00007f6ccb219000)\n    /lib64/ld-linux-x86-64.so.2 (0x00007f6ccc440000)\n    libdl.so.2 => /usr/lib64/libdl.so.2 (0x00007f6ccb014000)\n    libutil.so.1 => /usr/lib64/libutil.so.1 (0x00007f6ccae11000)\n```\n\nSame problem, it cannot load `libopenblas.so`.  So let's tell Spack to load it explicitly and try again:\n\n```\n[me@ankeli spack2]$ spack load openblas\n[me@ankeli spack2]$ ldd `spack location -i py-numpy`/lib/python3.5/site-packages/numpy-1.11.0-py3.5-linux-x86_64.egg/numpy/core/multiarray.cpython-35m-x86_64-linux-gnu.so\n    linux-vdso.so.1 =>  (0x00007ffc78d2c000)\n    libopenblas.so.0 => /home/me/spack2/opt/spack/linux-x86_64/gcc-4.9.3/openblas-0.2.17-pw6rmlom7apfsnjtzfttyayzc7nx5e7y/lib/libopenblas.so.0 (0x00007fb61f48e000)\n    libm.so.6 => /lib64/libm.so.6 (0x00007fb61f175000)\n    libpython3.5m.so.1.0 => /home/me/spack2/opt/spack/linux-x86_64/gcc-4.9.3/python-3.5.1-5q5rsrtjld4u6jiicuvtnx52m7tfhegi/lib/libpython3.5m.so.1.0 (0x00007fb61ec4f000)\n    libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fb61ea32000)\n    libc.so.6 => /lib64/libc.so.6 (0x00007fb61e671000)\n    /lib64/ld-linux-x86-64.so.2 (0x00007fb620878000)\n    libgfortran.so.3 => /home/me/spack/opt/spack/linux-x86_64/gcc-4.8.5/gcc-4.9.3-layphctulnk3omsbjpzftqv6dlxpfe3d/lib64/libgfortran.so.3 (0x00007fb61e352000)\n    libdl.so.2 => /usr/lib64/libdl.so.2 (0x00007fb61e14d000)\n    libutil.so.1 => /usr/lib64/libutil.so.1 (0x00007fb61df4a000)\n    libquadmath.so.0 => /home/me/spack/opt/spack/linux-x86_64/gcc-4.8.5/gcc-4.9.3-layphctulnk3omsbjpzftqv6dlxpfe3d/lib64/../lib64/libquadmath.so.0 (0x00007fb61dd0c000)\n    libgcc_s.so.1 => /home/me/spack/opt/spack/linux-x86_64/gcc-4.8.5/gcc-4.9.3-layphctulnk3omsbjpzftqv6dlxpfe3d/lib64/../lib64/libgcc_s.so.1 (0x00007fb61daf6000)\n```\n\nOK, now it works with `ldd`.  And try with Python:\n\n```\n[me@ankeli spack2]$ python3 -c 'import numpy'\n```\n\nRemember... LD_LIBRARY_PATH is EVIL!!!\n",
    "performed_via_github_app": null
}